### DISCUSSION TITLE
No punctuation in dialogue box and not able to correctly slow down the automatic speech recognition

### DISCUSSION THREAD

**Discussion by Mathijs1985** [deepgram/community#966] [General help] [answered]
I'm using the Deepgram API as the STT solution for my chat interface. But I have been having 2 problems. Problem 1 is the absence of punctuation in the transcribed text. I've tried fixing it in the configuration settings, but I'm still having the issue. Problem 2 is with the automatic speech recognition. The user's speech input is being registered too fast. I've tried adjusting the audio input rate, but the results I got were not optimal. Is there anyone who could help me with this?

**Comment by deepgram-community** [comment]
Hey there! It looks like you haven't connected your GitHub account to your Deepgram account. You can do this at https://community.deepgram.com - being verified through this process will allow our team to help you in a much more streamlined fashion.

**Reply by Mathijs1985** [reply]
![Schermafbeelding 2024-10-19 223421](https://github.com/user-attachments/assets/1819b859-fb56-47b7-9a95-808cb4fd62cb)

This is the message I get when I try to link my account, for Discord I did not have this problem.

**Comment by deepgram-community** [comment]
It looks like we're missing some important information to help debug your issue. Would you mind providing us with the following details in a reply?

- The programming language you are working in (e.g. JavaScript, Python).
- A request ID that triggered your error or issue.

**Reply by Mathijs1985** [reply]
The programming language is JavaScript and Python, the interface was built by a developer. I don't have 'request ID'. I'm basically just having issues getting these two functions to work properly (automatic speech recognition and punctuation in the transcribed text)

**Comment by deepgram-community** [comment, marked as answer]
I'm sorry to hear that you're having trouble with the automatic speech recognition speed and punctuation in the transcribed text. Let's try to address these issues one by one.

For the punctuation issue, Deepgram's Dictation feature might be what you need. It converts spoken dictation commands into their corresponding punctuation marks. To enable Dictation, set the `dictation` parameter to `true` in your API request. Note that the `punctuate` parameter must also be enabled for Dictation to work. Here's an example of how to do this:

```bash
curl \
  --request POST \
  --header 'Authorization: Token YOUR_DEEPGRAM_API_KEY' \
  --header 'Content-Type: audio/wav' \
  --data-binary @youraudio.wav \
  --url '<https://api.deepgram.com/v1/listen?dictation=true&punctuate=true>'
```

As for the automatic speech recognition speed, it's hard to provide a specific solution without more details. However, you might want to check the audio input rate and ensure it's compatible with Deepgram's requirements. If the audio input rate is too high, it might cause the speech recognition to process too quickly.

If you're still having trouble, could you please provide more details about your implementation? For example, it would be helpful to know how you're using the Deepgram API (e.g., are you using the SDK or making direct API calls?), and any relevant code snippets. This will help us better understand your situation and provide more targeted assistance.
> Naomi

**Reply by Mathijs1985** [reply]
Thank you very much for your reply, and I'm in the process of trying to implement these suggestions. One of my issues is that I'm not very familiar with all the technical aspects and I have the tendency to employ developers for this. Hence, I've shared your feedback with someone I'm collaborating with, and I'm really hoping that the solutions work. 

At the moment I am using a send button on my interface, but one of the reasons I choose to go with Deepgram, was your automatic speech recognition, which really does improve the UX (at least, if implemented correctly). 

Just some quick insight, I'm a language teacher and I'm digitalising my courses using AI. I basically use Openai's GPT for the input and output, Deepgram for the STT, Google Neural Voices for TTS, and then an additional solution for speech graphics (a digital character 'avatar'). 
Right now my chat interface's dialogue box doesn't show punctuation and the STT is not automated. I got some advice through your Discord channel, too. And I'm really hoping that all this, combined with my own R&D, will give me a satisfactory solution. 

FYI I'm making direct API calls. 

Thanks again, 
Mathijs
